lib/charms/layer/apache_kafka.py

```diff
@@ -14,14 +16,29 @@ def __init__(self, dist_config=None):
         self.resources = {
             'kafka': 'kafka-%s' % utils.cpu_arch(),
         }
-        self.verify_resources = utils.verify_resources(*self.resources.values())
+
+    def verify_resources(self):
+        if hookenv.resource_get(self.resources['kafka']):
```

Comment:
This will also need to be encapsulated in a try/except block if you want to retain backwords compat with 1.25 deployments.